allensdk/brain_observatory/behavior/behavior_session.py

```diff
@@ -139,54 +233,82 @@ def from_nwb(cls, nwbfile: NWBFile) -> "BehaviorSession":
         raw_running_speed = RunningSpeed.from_nwb(nwbfile, filtered=False)
         running_speed = RunningSpeed.from_nwb(nwbfile)
         metadata = BehaviorMetadata.from_nwb(nwbfile)
-
-        return cls(
+        licks = Licks.from_nwb(nwbfile=nwbfile)
+        rewards = Rewards.from_nwb(nwbfile=nwbfile)
+        stimuli = Stimuli.from_nwb(nwbfile=nwbfile)
+        task_parameters = TaskParameters.from_nwb(nwbfile=nwbfile)
+        trials = TrialTable.from_nwb(nwbfile=nwbfile)
+        date_of_acquisition = DateOfAcquisition.from_nwb(nwbfile=nwbfile)
+
+        return BehaviorSession(
             behavior_session_id=behavior_session_id,
             stimulus_timestamps=stimulus_timestamps,
             running_acquisition=running_acquisition,
             raw_running_speed=raw_running_speed,
             running_speed=running_speed,
-            metadata=metadata
+            metadata=metadata,
+            licks=licks,
+            rewards=rewards,
+            stimuli=stimuli,
+            task_parameters=task_parameters,
+            trials=trials,
+            date_of_acquisition=date_of_acquisition
         )
 
     @classmethod
-    def from_nwb_path(cls, nwb_path: str):
+    def from_nwb_path(cls, nwb_path: str) -> "BehaviorSession":
+        """
+
+        Parameters
+        ----------
+        nwb_path
+            Path to nwb file
+
+        Returns
+        -------
+        An instantiation of a `BehaviorSession`
+        """
         with pynwb.NWBHDF5IO(str(nwb_path), 'r') as read_io:
             nwbfile = read_io.read()
             return cls.from_nwb(nwbfile=nwbfile)
 
-    def to_json(self) -> dict:
-        pass
+    def to_nwb(self, add_metadata=True) -> NWBFile:
+        """
+
+        Parameters
+        ----------
+        add_metadata
+            Set this to False to avoid writing metadata multiple times.
+            An example use case is in BehaviorOphysExperiment, which builds on
+            behavior metadata. Metadata needs to be written at once to nwb, and
```

Comment:
What I meant to say is it requires something like
```
nwb.add_metadata(...)
```

It can't be separated like
```
nwb.add_metadata(...)
nwb.add_metadata(...)
```

That's what I mean by "written at once". Maybe "serialized at once" is better. Is it still unclear?

Reply:
"written all at once" I think would be clearer, if more verbose.